src/main/java/org/testeditor/testenv/vagrant/AbstractVagrantMojo.java

```diff
@@ -0,0 +1,104 @@
+/*******************************************************************************
+ * Copyright (c) 2012 - 2016 Signal Iduna Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ *******************************************************************************/
+package org.testeditor.testenv.vagrant;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Parameter;
+
+public abstract class AbstractVagrantMojo extends AbstractMojo {
+
+	@Parameter(property = "vagrantFileDir", defaultValue = "vagrant-vm")
+	private File vagrantFileDir;
+
+	public File getVagrantFileDir() {
+		return vagrantFileDir;
+	}
+
+	protected int executeVagrantCommand(String... command) throws MojoExecutionException {
+		ProcessBuilder builder = new ProcessBuilder(command);
+		builder.directory(getVagrantFileDir());
+		getLog().debug(getVagrantFileDir().toString());
+		try {
+			Process upPrc = builder.start();
+			createAndRunLoggerOnStream(upPrc.getInputStream(), false);
+			createAndRunLoggerOnStream(upPrc.getErrorStream(), false);
+			while (isAlive(upPrc)) {
+				Thread.sleep(100);
+			}
+			return upPrc.exitValue();
+		} catch (IOException e) {
+			getLog().error(e);
+			throw new MojoExecutionException("", e);
+		} catch (InterruptedException e) {
```

Comment:
We should use a multi-catch here and could use the message parameter to output some useful context such as the commands that we tried to execute on vagrant.